CharlieBackend.Api.UnitTes/ScheduleServiceTests.cs

```diff
@@ -0,0 +1,415 @@
+            _unitOfWorkMock.Setup(x => x.MentorRepository).Returns(_mentorRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.StudentGroupRepository).Returns(_studentGroupRepositoryMock.Object);
+        }
+
+        private void Initialize(CreateScheduleDto createScheduleDto)
+        {
+            _scheduleRepositoryMock.Setup(x => x.AddRange(new List<ScheduledEvent>()
+            {
+                new ScheduledEvent
+                {
+                    ThemeId = 5,
+                    StudentGroupId = 3,
+                    MentorId = 1
+                }
+            }
+            ));
+
+            var eventOccuranceRepositoryMock = new Mock<IEventOccurrenceRepository>();
+
+            if (createScheduleDto != null)
+            {
+                eventOccuranceRepositoryMock.Setup(x => x.Add(
+                    new EventOccurrence
+                    {
+                        StudentGroupId = 3,
+                        EventStart = createScheduleDto.Range.StartDate,
+                        Pattern = createScheduleDto.Pattern.Type
+                    }
+                    ));
+            }
+
+            _unitOfWorkMock.Setup(x => x.EventOccurrenceRepository).Returns(eventOccuranceRepositoryMock.Object);
+        }
+
+        [Fact]
+        public async Task CreateScheduleAsync_ValidScheduleRequest_ShouldReturnEventOccurrenceDTO()
+        {
+            //Arrange
+            var createScheduleDto = new CreateScheduleDto
+            {
+                Pattern = new PatternForCreateScheduleDTO
+                {
+                    Type = PatternType.Daily,
+                    Interval = 1,
+                    DaysOfWeek = new List<DayOfWeek> { DayOfWeek.Monday },
+                },
+
+                Range = new OccurenceRange
+                {
+                    StartDate = DateTime.Parse("Jan 1, 2021"),
+                    FinishDate = DateTime.Parse("Feb 1, 2021")
+                },
+
+                Context = new ContextForCreateScheduleDTO
+                {
+                    GroupID = 3,
+                    MentorID = 1,
+                    ThemeID = 5
+                }
+            };
+
+            EventOccurrence expected = new EventOccurrence
+            {
+                Pattern = createScheduleDto.Pattern.Type,
+                StudentGroupId = createScheduleDto.Context.GroupID,
+                EventStart = createScheduleDto.Range.StartDate,
+                EventFinish = createScheduleDto.Range.FinishDate.Value,
+                Storage = EventOccuranceStorageParser.GetPatternStorageValue(createScheduleDto.Pattern)
+            };
+
+            _themeRepositoryMock.Setup(x => x.IsEntityExistAsync(5)).ReturnsAsync(true);
+
+            _mentorRepositoryMock.Setup(x => x.IsEntityExistAsync(1)).ReturnsAsync(true);
+
+            _studentGroupRepositoryMock.Setup(x => x.IsEntityExistAsync(3)).ReturnsAsync(true);
```

Comment:
Perhaps we could improve readability even more by removing these magic numbers (that are also present in lines 93-95) and declaring them as variables/constants:
```cs
long existentGroupId = 3;
long existentMentorId = 1;
long existentThemeId = 5;
```

This way we also will be able to change them in one place instead of looking through the whole method and updating the magic numbers and introducing new mistakes.

Reply:
All changes were implemented in https://github.com/ita-social-projects/WhatBackend/pull/406/commits/1b701da3ad50209af1dd4bebd6f32224769e61c5